### PR TITLE
Add icon and link params to `gr.Button`

### PR DESCRIPTION
**Pull request by hannahblair**
## Description

This PR allows us to add a link and icon to the `gr.Button` component. The default icon placement is central alongside the text:

Screenshot 2023-08-03 at 16 03 37

If a different icon placement is desired, we can use custom css. 

Screenshot 2023-08-03 at 16 15 10

--- 
Icon Demo

```
import gradio as gr

with gr.Blocks(css = """button {justify-content: left !important;}""") as demo:
    gr.Button(value="Login with HF", icon='https://huggingface.co/front/assets/huggingface_logo-noborder.svg')
    
demo.launch()
```

--- 
Link Demo

```
import gradio as gr

with gr.Blocks() as demo:
    gr.Button(value="Login with HF", link="https://huggingface.co/login")
    
demo.launch()
```

Closes: #5060 

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

**Comment by abidlabs** [comment]
Very cool! cc @Wauplin

**Comment by Wauplin** [comment]
Awesome! Love this work @hannahblair !

**Comment by abidlabs** [comment]
Noticed something interesting while reviewing @hannahblair:

If I add a `link`, the width of the `gr.Button()` changes:

```py
import gradio as gr

with gr.Blocks() as demo:
    gr.Button(icon="cheetah.jpg")
    
demo.launch()
```

produces this:

image

whereas:

```py
import gradio as gr

with gr.Blocks() as demo:
    gr.Button(icon="cheetah.jpg", link="https://www.google.com")
    
demo.launch()
```

produces this:

image

File: js/button/static/LinkWrapper.svelte [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<script lang="ts">
+	export let link: string | null;
+</script>
+
+{#if link && link.length > 0}
+	<a href={link} target="_blank" rel="noreferrer">
```
Comment (abidlabs) [inline js/button/static/LinkWrapper.svelte:6]: Similar to #4536, we shouldn't open anchor links in a new tab.

File: gradio/components/button.py [hunk 2]
```diff
@@ -67,11 +71,20 @@ def __init__(
         self.variant = variant
         self.size = size
 
+        if icon is not None:
+            self.icon = "/file=" + icon
+        else:
+            self.icon = icon
```
Comment (abidlabs) [inline gradio/components/button.py:77]: style nit
```suggestion
        self.icon = icon if icon is None else "/file=" + icon
```

**Comment by abidlabs** [comment]
Nice @hannahblair! Just a few nits above and then should be good to go

**Review by abidlabs — APPROVED** [review]

**Comment by Wauplin** [comment]
I'm sorry but I can't seem to make icon button work locally. What I did so far:
- `git checkout main` + `git pull`
- create a `app.py` file with
```py
import gradio as gr

with gr.Blocks(css="""button {justify-content: left !important;}""") as demo:
    gr.Button(value="Login with HF", icon="https://huggingface.co/front/assets/huggingface_logo-noborder.svg")

demo.launch()
```
- run `python app.py`

And I only get a simple button with text but without the logo. I guess I'm missing something but I don't know what. I also tried `bash scripts/build_frontend.sh` in case something has to be rebuilt but it fails with an error related to gradio js client:

```
[vite]: Rollup failed to resolve import "semiver" from "/home/wauplin/projects/gradio/client/js/src/client.ts".
```

Any way I could get that working? I'm looking forward to integrate it in the "Sign in with HF" PR to get nice icons :smile:

**Comment by abidlabs** [comment]
That code is working for me @Wauplin. The most likely culprit is that you need to build the frontend first. 

Can you try running `pnpm i` from the root `gradio` directory (the same directory where you run `bash scripts/build_frontend.sh`) and then re-run `bash scripts/build_frontend.sh`

**Comment by Wauplin** [comment]
Thanks both! I finally made it work :smile: I deleted `node_modules/` folder and reinstalled everything which fixed my issue. 
And I can confirm that I have my HF-button working now! :hugs:

**Comment by shinetzh** [comment]
when use gr.update(icon=...), the icon of the button went wrong, like this
image

the original button is this:
image

**Comment by muge-birlik** [comment]
When clicked on a button with a link, Is it possible to open the link in a new tab?

**Comment by abidlabs** [comment]
@muge-birlik that is already the case by default for any links that are not anchor links in the same page. 

e.g. try with:

```py
import gradio as gr

with gr.Blocks() as demo:
    textbox = gr.Textbox()
    with gr.Row():
        b1 = gr.Button(link="https://www.google.com", scale=2, min_width=0)
        b2 = gr.Button(link="#", scale=1, min_width=0)
    b1.click(lambda :"hi", None, textbox)
    
demo.launch()
```

**Pull request merged**